chain/network/src/peer/peer_actor.rs

```diff
@@ -251,22 +271,12 @@ impl PeerActor {
     }
 
     fn fetch_client_chain_info(&self, ctx: &mut Context<PeerActor>) {
```

Comment:
As of your previous comment, it would be nice to have genesis_id directly in the PeerManagerState, initialized in the PeerManagerActor constructor. I'd like to add the genesis_id to the NetworkConfig (which is an argument to PeerManagerActor constructor), but it seems that computing genesis block hash is terribly convoluted: it requires constructing Chain, which in turn requires ChainGenesis and RuntimeAdapter. What do you thing would be the easiest way to directly compute a genesis_id from Genesis object?